src/main/java/com/google/sps/servlets/CaptionService.java

```diff
@@ -0,0 +1,18 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class CaptionService {
```

Comment:
You had a comment about testing CaptionService. Hmm... maybe you don't need CaptionService at all if you can somehow mock the openStream method for URL class to return an stream that contains XML content you would like to test on. What do you think?

Reply:
That was my first idea too! unfortunately, Mockito can't mock final objects such as a URL. VSE and other people suggested using a wrapper class like this

Reply:
Looks like you can in the version of Mockito you are using. Not sure if even worth the effort anymore.
https://stackoverflow.com/questions/14292863/how-to-mock-a-final-class-with-mockito

Reply:
Lol I wonder why I didn't find it but thank you! I have included the plugin I am sure it will be useful for testing APIs, because some objects returned by the YouTube API (i.e. `CommentThread`) are final objects too.